### ISSUE TITLE
According to the paper, these two summands should be exchanged, should they not?

### ISSUE THREAD

**Issue by CDitzel** [karpathy/deep-vector-quantization#2]
https://github.com/karpathy/deep-vector-quantization/blob/bcc1c38bec754caf0398c4cd054f34a963d348b4/model.py#L58

I believe, the commitment loss term features the stop-gradient op for the quantized latents

`diff = (z_q- z_e.detach()).pow(2).mean() + commitment_cost * (z_q.detach()  - z_e).pow(2).mean() `

**Comment by karpathy** [comment]
thank you good catch! fixed with https://github.com/karpathy/deep-vector-quantization/commit/b39653a8bb1d36bcd81348ee5b3c9476cd7ca546

re-ran and it converges to about the same result, closing.
